### ISSUE TITLE
[Bug] [Shapes] Drawing inconsistencies

### ISSUE THREAD

**Issue by opcodewriter** [xamarin/Xamarin.Forms#11498]
### Description

On Android, there are at least two issues with the Rectangle shape (not sure if it's just this shape), please see the screenshots below:

- It's drawn with some border/margin. Note `StrokeThickness` is set to zero.
- The corners don't look straight, they seem rounded (it should look like for the BoxView). On iOS, they look perfectly straight.

Please test the other shapes too. 
My suggestion is to create a StackLayout with all the possible Shapes stacked, and create a screenshot for Android and iOS and compare them to check if they look the same.

### Steps to Reproduce

```
    <StackLayout BackgroundColor="Red"
                 Spacing="0">
        <Rectangle HeightRequest="30"
                   Fill="Black"
                   StrokeThickness="0" />
        <BoxView HeightRequest="30"
                 BackgroundColor="Black" />
    </StackLayout>
```

### Expected Behavior

No border/margin and straight corners.

### Actual Behavior

Android (from the XAML preview, but in emulator it looks the same):

![image](https://user-images.githubusercontent.com/15019084/88030017-65d61e00-cb43-11ea-9737-565c369908f6.png)

iOS (looks good):

![image](https://user-images.githubusercontent.com/15019084/88030572-10e6d780-cb44-11ea-90c3-cab05667b594.png)

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

**Comment by hartez** [comment]
[_11498 Repro.zip](https://github.com/xamarin/Xamarin.Forms/files/4974541/_11498.Repro.zip)

**Comment by opcodewriter** [comment]
Is there really any plans to address this?

**Comment by Inrego** [comment]
I think the border/margin issue may be due to strokes being set even when StrokeThickness is set to 0. Try to give your stroke a color, then it's more visible. Possibly, settings stroke color to the same as fill will give you the desired result.

**Comment by opcodewriter** [comment]
@Inrego I don't think the StrokeThickness is still set when it's set to zero. It's 1 by default, but if you set it to 0, it *should* work...

> Possibly, settings stroke color to the same as fill will give you the desired result.

Right, but this isn't a good solution in my case, the stroke makes the shape bigger than it should be..

**Comment by Inrego** [comment]
I've experienced that if I set StrokeThickness to 0, then there's still a visible stroke (if the stroke has a color).
I think if you set StrokeThickness to 0, and Stroke to same color as fill, it will be correct size (though I'm not sure, only guessing).

**Comment by Inrego** [comment]
But my experiences are only from Ellipse. Maybe Rectangle is different.

**Comment by opcodewriter** [comment]
Can the Forms team please confirm this issue has been acknowledge and it will be fixed?

Thanks!

**Comment by jsuarezruiz** [comment]
@opcodewriter Yes, I self assigned the issue and will review it in the next days.
